### PR TITLE
Compatibility and firmware requirements.

### PR DESCRIPTION
**Pull request by johnwilkins**
Fixes: [knideploy-2088](https://issues.redhat.com/browse/knideploy-2088), [telcodocs-108](https://issues.redhat.com/browse/telcodocs-108), [telcodocs-109](https://issues.redhat.com/browse/telcodocs-109) and [telcodocs-131](https://issues.redhat.com/browse/telcodocs-131)

@ahardin-rh @iurygregory 

Signed-off-by: John Wilkins <jowilkin@redhat.com>

### CODE REVIEW

File: modules/ipi-install-firmware-requirements-for-installing-with-virtual-media.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+|====
+
+[NOTE]
+====
+Refer to the hardware documentation for the nodes or contact the hardware vendor for information on updating the firmware.
+
+There are no known firmware limitations for HP servers.
+
+For Dell servers, ensure the {product-title} cluster nodes have AutoAttach Enabled through the iDRAC console. The menu path is: *Configuration* -> *Virtual Media* -> *Attach Mode* -> *AutoAttach* . With iDRAC 9 firmware version `04.40.00.00`, the Virtual Console Plugin defaults to `eHTML5`, which causes problems with the *InsertVirtualMedia* workflow. Set the Plugin to `HTML5` to avoid this issue.
```
Comment (iurygregory) [inline modules/ipi-install-firmware-requirements-for-installing-with-virtual-media.adoc:29]: Maybe we should mention the path to update the Virtual Console Plugin?

Comment (johnwilkins) [reply modules/ipi-install-firmware-requirements-for-installing-with-virtual-media.adoc]: Added.

**Review by iurygregory — APPROVED** [review]
LGTM, Ty!

File: modules/ipi-install-bmc-addressing.adoc [hunk 2]
```diff
@@ -96,7 +94,9 @@ platform:
 
 [NOTE]
 ====
-Currently RedFish is only supported on Dell with iDRAC firmware version `4.20.20.20` or higher for installer-provisioned installations of {product-title} on bare metal deployments.
+Currently RedFish is only supported on Dell with iDRAC firmware versions `4.20.20.20` through `04.40.00.00` for IPI on Bare metal deployments. There is a known issue with version `04.40.00.00`. With iDRAC 9 firmware version `04.40.00.00`, the Virtual Console Plugin defaults to `eHTML5`, which causes problems with the *InsertVirtualMedia* workflow. Set the Plugin to `HTML5` to avoid this issue. The menu path is: *Configuration* -> *Virtual console* -> *Plug-in Type* -> *HTML5* .
```
Comment (ahardin-rh) [inline modules/ipi-install-bmc-addressing.adoc:97]: s/Bare metal/bare meta
s/Plugin/plug-in

File: modules/ipi-install-bmc-addressing.adoc [hunk 3]
```diff
@@ -134,11 +175,13 @@ platform:
...
           username: <user>
           password: <password>
+          disableCertificateVerification: True
 ----
 
+
 [NOTE]
 ====
-`idrac-virtualmedia` requires iDRAC firmware version 4.20.20.20 or higher.
+Currently RedFish is only supported on Dell with iDRAC firmware versions `4.20.20.20` through `04.40.00.00` for IPI on Bare metal deployments. There is a known issue with version `04.40.00.00`. With iDRAC 9 firmware version `04.40.00.00`, the Virtual Console Plugin defaults to `eHTML5`, which causes problems with the *InsertVirtualMedia* workflow. Set the Plugin to `HTML5` to avoid this issue. The menu path is: *Configuration* -> *Virtual console* -> *Plug-in Type* -> *HTML5* .
```
Comment (ahardin-rh) [inline modules/ipi-install-bmc-addressing.adoc:184]: Plug-in

File: modules/ipi-install-creating-a-disconnected-registry.adoc [hunk 4]
```diff
@@ -7,12 +7,12 @@
 
 = Creating a disconnected registry (optional)
 
-In some cases, you might want to install an OpenShift KNI cluster using a local copy of the installation registry. This could be for enhancing network efficiency because the cluster nodes are on a network that does not have access to the internet.
+In some cases, you might want to install an Openshift KNI cluster using a local copy of the installation registry. This could be for enhancing network efficiency because the cluster nodes are on a network that does not have access to the internet.
```
Comment (ahardin-rh) [inline modules/ipi-install-creating-a-disconnected-registry.adoc:10]: OpenShift is correct

File: modules/ipi-install-firmware-requirements-for-installing-with-virtual-media.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+|====
+
+[NOTE]
+====
+Refer to the hardware documentation for the nodes or contact the hardware vendor for information on updating the firmware.
+
+There are no known firmware limitations for HP servers.
+
+For Dell servers, ensure the {product-title} cluster nodes have AutoAttach Enabled through the iDRAC console. The menu path is: *Configuration* -> *Virtual Media* -> *Attach Mode* -> *AutoAttach* . With iDRAC 9 firmware version `04.40.00.00`, the Virtual Console Plugin defaults to `eHTML5`, which causes problems with the *InsertVirtualMedia* workflow. Set the Plugin to `HTML5` to avoid this issue. The menu path is: *Configuration* -> *Virtual console* -> *Plug-in Type* -> *HTML5* .
```
Comment (ahardin-rh) [inline modules/ipi-install-firmware-requirements-for-installing-with-virtual-media.adoc:29]: Plug-in

**Comment by ahardin-rh** [comment]
@iurygregory Is this work targeting 4.7 or 4.8? I notice that the associated JIRAs are for 4.8 so just want to confirm before merging. Thanks!

**Comment by iurygregory** [comment]
@ahardin-rh we need this documentation in 4.7, we will have code to verify the support for virtual media in 4.8. I think we will also need the information about firmware in 4.6 docs, because virtual media support was added in 4.6

**Comment by ahardin-rh** [comment]
Thank you!

**Comment by ahardin-rh** [comment]
@johnwilkins You might need to look into what content needs backported to 4.6 based off of Iury's comment above. I will get this in for 4.7. Than you, all!

**Pull request merged**